Client/Frontend/Browser/BackForwardListView.swift

```diff
@@ -0,0 +1,196 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import WebKit
+import Storage
+import SnapKit
+
+class BackForwardListView: UIView, UITableViewDataSource, UITableViewDelegate {
+    
+    enum BackForwardType {
+        case Forward
+        case Current
+        case Backward
+    }
+    
+    struct BackForwardViewUX {
+        static let RowHeight = 50
+    }
+    
+    private var profile: Profile?
+    private var sites = [String: Site]()
+    private var tableView: UITableView!
+    private var isPrivate: Bool!
+    private var dismissing = false
+    var tabManager: TabManager!
+    var listData = [(item:WKBackForwardListItem, type:BackForwardType)]()
+    
+    init(profile: Profile, backForwardList: WKBackForwardList, isPrivate: Bool) {
+        super.init(frame: CGRect.zero)
+        self.backgroundColor = UIColor.clearColor()
+        
+        self.profile = profile
+        self.isPrivate = isPrivate
+        
+        tableView = UITableView()
+        tableView.separatorStyle = .None
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.tableFooterView = UIView(frame: CGRectMake(0, 0, tableView.frame.size.width, 1))
```

Comment:
Nit: Is this the trick to remove the lines past the bottom? I think you can just use `UIView()` with no frame.